File: Web/angularjs-app/ClassWork/SinglePageApp/StudentAPI-spa/studentService.js
angular.module('studentModule')
    .factory('studentFactory', ['$http','$log', function ($http,$log) {
        var factory = {}

        factory.Student = function(rollNo, name, age, email, date, isMale) {
            this.rollNo = rollNo
            this.name = name
            this.age = age
            this.email = email
            let formattedDate = date.getFullYear() + "-" + ("0" + (date.getMonth() + 1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2);
            this.date = formattedDate
            this.isMale = isMale
        }
       
        factory.getStudents = function () {
          return  $http.get("http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students")
        }

        factory.addStudent = function(data){
            return $http.post("http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students",data)
        }

        factory.getStudentById = function(studentId){
            return  $http.get("http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students/"+studentId)
        }

        factory.updateStudent = function(updatedStudent){
            return $http.put("http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students/"+updatedStudent.id,updatedStudent)
        }

        factory.deleteStudent = function(studentId){
            return $http.delete("http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students/"+studentId)
        }

        return factory
    }])

File: Spring/struts-spring-hibernate-app/src/main/java/com/techlabs/service/StudentService.java
package com.techlabs.service;

import java.util.List;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.techlabs.model.Student;
import com.techlabs.repository.StudentRepository;

@Service()
public class StudentService {
	
	@Autowired
	private StudentRepository repository;
	
	public StudentService() {
		System.out.println("Inside Student Service");
	}
	
	public List<Student> getStudents(){
		return repository.getStudents();
	}
	
	public void addStudent(int rollNo, String name, int age, String email) {
		repository.addStudent(rollNo, name, age, email);
	}
	
	public Student getStudent(String id) {
		return repository.getStudent(id);
	}
	
	public void updateStudent(String id, int rollNo, String name, int age, String email) {
		repository.updateStudent(id, rollNo, name, age, email);
	}
	
	public void deleteStudent(String id) {
		repository.deleteStudent(id);
	}

}
